vivisect/analysis/elf/libc_start_main.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import envi
+import logging
+
+import vivisect
+import vivisect.impemu.monitor as viv_imp_monitor
+
+logger = logging.getLogger(__name__)
+
+
+def analyzeFunction(vw, funcva):
+    '''
+    search through all calls, looking for a call to an import named __libc_start_main
+    then check for arg0
+    '''
+    global emu, emumon, cconv
+    try:
+        emu = vw.getEmulator()
+        emumon = AnalysisMonitor(vw, funcva)
+
+        emu.setEmulationMonitor(emumon)
+        emu.runFunction(funcva, maxhit=1)
+
+        if not emumon.success:
+            return
+
+        api = vw.getFunctionApi(emumon.startmain)
+        cconv = emu.getCallingConvention(api[vivisect.API_CCONV])
+
+        args = cconv.getCallArgs(emu, 6)
+        print args
+        vw.addEntryPoint(args[0])
+        vw.makeFunction(args[0])
+
+    except Exception as e:
+        sys.excepthook(*sys.exc_info())
+
+
+
+def analyze(vw):
+    global emu, emumon
+    cg = vw.getCallGraph()
+
+    for va, name in vw.getNames():
+        if name == '__libc_start_main_%.8x' % va:
+            for xfr, xto, xtype, xtinfo in vw.getXrefsTo(va):
+                logger.info("0x%x -> 0x%x", xfr, xto)
+                arg0va = vw.getLocation(xfr-1)[0]
```

Comment:
this is totally broken, since we're assuming that the previous instruction is a "push main".  i apologize, this file wasn't actually ready for review.  

i like the emulation approach best, but after implementing it, i realized that - with proper sequencing - the global approach was better.... and tried a different approach.  i believe when we've stewed on it, the emulation approach will emerge as the better option.  the only drawback is that the architecture *must* implement an emulator to get the functionality, but as soon as the latest ARM and PPC branches are merged in, i don't think that will be a problem.  

we can probably just make the analyze() function call analyzeFunction() with the appropriately identified functions based on xrefs.